src/BreakpointViewer.cpp

```diff
@@ -189,6 +193,10 @@ void BreakpointViewer::removeBreakpoint(BreakpointRef::Type type, int row, bool
 
 	auto* command = new Command(cmdStr,
 		[this, type, row, id, logical] (const QString& /*result*/) {
+			// breakpoints will be reloaded anyway if update events are supported
+			if (debugSession.hasDebugUpdates())
+				return;
+
```

Comment:
Both here and above (line 143).

This is correct. But my preference is to not have two implementations for the same thing (either do the update manually or rely on the debug-update event). One reason for this is that we now have two different code paths, and only one path gets tested (openMSX developers typically only test the code path for the newest openMSX version).

in other words: I think it's OK to manually add the breakpoint, and then soon after have the list completely refreshed. I mean as a tradeoff with having two code paths (of which one path becomes untested).

Personally I would leave out these two if-statements. But fine if you really want them (maybe for some reason that I'm currently overlooking).

(detail, style)  We prefer to either:
* Put the full if-statement on a single line:   `if (a == 1) return;`
* Or use multiple lines, but then put braces around the statement in the then-block:
```
if (a == 1) {
    return;
}
```